avm/res/web/site/slot/extension/main.bicep

```diff
@@ -0,0 +1,46 @@
+metadata name = 'Site Deployment Extension '
+metadata description = 'This module deploys a Site extension for MSDeploy.'
+
+@description('Conditional. The name of the parent site resource. Required if the template is used in a standalone deployment.')
+param appName string
+
+@description('Conditional. The name of the parent web site slot. Required if the template is used in a standalone deployment.')
+param slotName string
+
+@description('Optional. The name of the extension.')
+@allowed([
+  'MSDeploy'
+])
+param name string = 'MSDeploy'
+
+@description('Optional. The kind of extension.')
+@allowed([
+  'MSDeploy'
+])
+param kind string = 'MSDeploy'
+
+@description('Optional. Sets the properties.')
+param properties resourceInput<'Microsoft.Web/sites/extensions@2024-04-01'>.properties?
+
+resource app 'Microsoft.Web/sites@2024-04-01' existing = {
+  name: appName
+
+  resource slot 'slots' existing = {
+    name: slotName
+  }
+}
+resource msdeploy 'Microsoft.Web/sites/slots/extensions@2024-04-01' = {
```

Comment:
Hey @tsc-buddy & @pankajagrawal16,
this is the interesting case where the original resource type was 

https://github.com/Azure/bicep-registry-modules/blob/85c473a7d68c82e7e36462221f1f74e568a3f986/avm/res/web/site/slot/extensions--msdeploy/main.bicep#L13

Even though this module is a child of the `slot` child-module. Hence the update to <code>Microsoft.Web/sites/<it><b>slots</b></it>/extensions</code>. Let me know what you think and if this was on purpose.

For the other folders it seems correct.

Reply:
Good Find. @tsc-buddy Any insights on this? This cannot be intentional ?